### PR TITLE
fix (github): handle create events, delete events

### PR DESCRIPTION
**Pull request by fsamin**

### CODE REVIEW

File: engine/api/auth/githubclient.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package auth
+
+import (
+	"net/http"
+
+	"github.com/go-gorp/gorp"
+	"github.com/ovh/cds/engine/api/context"
```
Comment (sguiheux) [inline engine/api/auth/githubclient.go]: need empty line between 6 and 7

File: engine/api/auth/githubclient.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+)
+
+//GithubConfig handles all config to connect to the github
+type GithubConfig struct {
+	ClientID     string
+	ClientSecret string
+}
+
+//GithubCLient is a github impl
+type GithubCLient struct{}
```
Comment (sguiheux) [inline engine/api/auth/githubclient.go]: GithubClient

File: engine/api/poller/executer.go [hunk 3]
```diff
@@ -6,6 +6,8 @@ import (
 
 	"github.com/go-gorp/gorp"
 
+	"database/sql"
```
Comment (sguiheux) [inline engine/api/poller/executer.go]: move to the top

File: engine/api/poller/executer.go [hunk 4]
```diff
@@ -125,7 +127,24 @@ func executerProcess(tx gorp.SqlExecutor, p *sdk.RepositoryPoller, e *sdk.Reposi
 		return nil, err
 	}
 
-	e.PushEvents, pollingDelay, err = client.PushEvents(p.Application.RepositoryFullname, p.DateCreation)
+	var events = []interface{}{}
+	events, pollingDelay, err = client.GetEvents(p.Application.RepositoryFullname, p.DateCreation)
```
Comment (sguiheux) [inline engine/api/poller/executer.go]: MIssing if err != nil

Comment (fsamin) [inline engine/api/poller/executer.go]: done

**Pull request merged**